network/nxos/nxos_portchannel.py

```diff
@@ -0,0 +1,742 @@
+            commands.append('interface {0}'.format(member))
+            commands.append('channel-group {0} mode {1}'.format(
+                existing['group'], proposed['mode']))
+
+    return commands
+
+
+def get_commands_to_remove_members(proposed, existing, module):
+    try:
+        proposed_members = proposed['members']
+    except KeyError:
+        proposed_members = []
+
+    try:
+        existing_members = existing['members']
+    except KeyError:
+        existing_members = []
+
+    members_to_remove = list(set(existing_members).difference(proposed_members))
+    commands = []
+    if members_to_remove:
+        for member in members_to_remove:
+            commands.append('interface {0}'.format(member))
+            commands.append('no channel-group {0}'.format(existing['group']))
+
+    return commands
+
+
+def get_commands_if_mode_change(proposed, existing, group, mode, module):
+    try:
+        proposed_members = proposed['members']
+    except KeyError:
+        proposed_members = []
+
+    try:
+        existing_members = existing['members']
+    except KeyError:
+        existing_members = []
+
+    try:
+        members_dict = existing['members_detail']
+    except KeyError:
+        members_dict = {}
+
+    members_to_remove = set(existing_members).difference(proposed_members)
+    members_with_mode_change = []
+    if members_dict:
+        for interface, values in members_dict.iteritems():
+            if (interface in proposed_members and
+                    (interface not in members_to_remove)):
+                if values['mode'] != mode:
+                    members_with_mode_change.append(interface)
+
+    commands = []
+    if members_with_mode_change:
+        for member in members_with_mode_change:
+            commands.append('interface {0}'.format(member))
+            commands.append('no channel-group {0}'.format(group))
+
+        for member in members_with_mode_change:
+            commands.append('interface {0}'.format(member))
+            commands.append('channel-group {0} mode {1}'.format(group, mode))
+
+    return commands
+
+
+def get_commands_min_links(existing, proposed, group, min_links, module):
+    commands = []
+    try:
+        if (existing['min_links'] is None or
+                (existing['min_links'] != proposed['min_links'])):
+            commands.append('interface port-channel{0}'.format(group))
+            commands.append('lacp min-link {0}'.format(min_links))
+    except KeyError:
+        commands.append('interface port-channel{0}'.format(group))
+        commands.append('lacp min-link {0}'.format(min_links))
+    return commands
+
+
+def flatten_list(command_lists):
+    flat_command_list = []
+    for command in command_lists:
+        if isinstance(command, list):
+            flat_command_list.extend(command)
+        else:
+            flat_command_list.append(command)
+    return flat_command_list
+
+
+def main():
+    argument_spec = dict(
```

Comment:
There are more options defined here than in the DOCUMENTAION

Any option specified in `extends_documentation_fragment: nxos` needs defining in `module_utils`, have a look at